docs/reference/scripting/using.asciidoc

```diff
@@ -1,198 +1,242 @@
-by a string instead of an object. This string contains the source of the script.
-
-Short form:
-
-[source,js]
-----------------------
-  "script": "ctx._source.my-int++"
-----------------------
-// NOTCONSOLE
+[[script-shorten-syntax]]
+=== Shorten your script
+Using syntactic abilities that are native to Painless, you can reduce verbosity
+in your scripts and make them shorter. For example, we can shorten the previous
+script and achieve the same goal:
 
-The same script in the normal form:
-
-[source,js]
-----------------------
-  "script": {
-    "source": "ctx._source.my-int++"
+[source,console]
+----
+GET my-index-000001/_search
+{
+  "script_fields": {
+    "my_doubled_field": {
+      "script": {
+        "lang":   "painless",
+        "source": "return doc['my_field'].value * params.get('multiplier');",
+        "params": {
+          "multiplier": 2
+        }
+      }
+    }
   }
-----------------------
-// NOTCONSOLE
-
-[discrete]
-[[modules-scripting-stored-scripts]]
-=== Stored scripts
+}
+----
+// TEST[s/^/PUT my-index-000001\n/]
 
-Scripts may be stored in and retrieved from the cluster state using the
-`_scripts` end-point.
+Let's look at the original version and see how what improvements it includes
+over this current iteration:
 
-If the {es} {security-features} are enabled, you must have the following 
-privileges to create, retrieve, and delete stored scripts:
+[source,console]
+----
+GET my-index-000001/_search
+{
+  "script_fields": {
+    "my_doubled_field": {
+      "script": {
+        "source": "doc['my_field'].value * params['multiplier']",
+        "params": {
+          "multiplier": 2
+        }
+      }
+    }
+  }
+}
+----
+// TEST[s/^/PUT my-index-000001\n/]
 
-* cluster: `all` or `manage`
+This version of the script removes several components and simplifies the syntax
+significantly:
 
-For more information, see <<security-privileges>>.
+* The `lang` declaration. Because Painless is the default language, you don't
+need to specify the language if you're writing a Painless script.
+* The `return` keyword. Painless automatically uses the final statement in a
+script (when possible) to produce a return value in a script context that
+requires one.
+* The `get` method, which is replaced with brackets `[]`. Painless
+uses a shortcut specifically for the `Map` type that allows us to use brackets
+instead of the lengthier `get` method.
+* The semicolon at the end of the `source` statement. Painless does not
+require semicolons for the final statement of a block. However, it does require
+them in other cases to remove ambiguity.
 
+Use this abbreviated syntax anywhere that {es} supports scripts.
 
 [discrete]
-==== Request examples
+[[script-stored-scripts]]
+=== Store and retrieve scripts
+You can store and retrieve scripts from the cluster state using the `_scripts`
+endpoint. Using stored scripts can help to reduce compilation time and make
+searches faster. Use the `{id}` path element `_scripts/{id}` to refer to a
```

Comment:
> Use the `{id}` path element in the stored scripts path `_scripts/{id}` to refer to a stored script.

> `_scripts/{id}`
is the full path

> `{id}` 
is an element of the path that identifies the stored script.